src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -47,6 +48,68 @@ protected static class ArtifactEventProperties
         }
     }
 
+        // Caller: PublishPipelineArtifact task
+    // Can be invoked from a build run or a release run should a build be set as the artifact. 
+    public class PublishPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
+    {
+        public override Guid Id => PipelineArtifactPluginConstants.PublishPipelineArtifactTaskId;
+
+        public static readonly Regex jobIdentifierRgx = new Regex("[^a-zA-Z0-9 - .]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        protected override async Task ProcessCommandInternalAsync(
+            AgentTaskPluginExecutionContext context, 
+            CancellationToken token)
+        {           
+            string artifactName = context.GetInput(ArtifactEventProperties.ArtifactName, required: false);
```

Comment:
`artifactName` is marked as "required" in the task.json ... but the logic here implies it is not. I actually prefer that it not be required, meaning we change task.json to not set a default (currently "drop") and for it to be optional.

Reply:
yes, we need to wait until the agent is rolled out and then set the required to false in the task. And the default should be an empty string I agree